sdk/identity/azure-identity/inc/azure/identity/managed_identity_credential.hpp

```diff
@@ -19,6 +19,29 @@ namespace Azure { namespace Identity {
     class ManagedIdentitySource;
   }
 
+  // This will move to Azure::Core.
+  /**
+   * @brief An Azure Resource Manager resource identifier.
+   */
+  class ResourceIdentifier {
```

Comment:
Currently, this type is a plain string wrapper in Identity, to avoid having a core dependency, for the beta release.

We plan to move this to Core, and consider:
- Add input validation in the ctor
- Once we have SDKs such as Azure.ResourceManager requiring more feature capability, add further APIs to "grow up" this type.

Reply:
Which packages beyond identity need this type?

Reply:
Currently none, but this is a core type that will be used by other SDK clients in the future like `Azure.ResourceManager` (and recommended to be added by @m-nash).

Keeping it in Identity would conflict at that time, in the future (which is what I am trying to avoid). As a concept, `ResourceIdentifier` is not an Identity concept, but a fairly common Azure concept.